examples/with-styled-components/pages/_document.js

```diff
@@ -0,0 +1,29 @@
+import Document, { Head, Main, NextScript } from 'next/document'
+import styleSheet from 'styled-components/lib/models/StyleSheet'
+
+export default class MyDocument extends Document {
+  static async getInitialProps (ctx) {
+    const renderPage = () => {
+      return ctx.renderPage()
+    }
+
+    const props = await Document.getInitialProps({ ...ctx, renderPage })
+    const style = styleSheet.rules().map(rule => rule.cssText).join('\n')
+    return { ...props, style }
+  }
```

Comment:
`Document.getInitialProps` includes the implementation of `styled-jsx` and `next/css`, so instead I think it should be like the following:

```js
static async getInitialProps ({ renderPage }) {
    const page = renderPage()
    const style = styleSheet.rules().map(rule => rule.cssText).join('\n')
    return { ...page, style }
}
```